docs/zkapps/writing-a-zkapp/feature-overview/actions-and-reducer.mdx

```diff
@@ -100,25 +99,37 @@ A zkApp can retrieve events and actions from one or more Mina archive nodes. If
 ### Reducer - API reference
 
 ```ts
-reducer = Reducer({ actionType: AsFieldElements<A> });
-
-this.reducer.dispatch(action: A): void;
+reducer = Reducer({ actionType: FlexibleProvablePure<Action> });
+
+this.reducer.dispatch(action: Action): void;
+
+this.reducer.reduce<State>(
+  actions: MerkleList<MerkleList<Action>>,
```

Comment:
Just checking if the double nested `MerkleList is the right usage?

Reply:
Here is the reduce definition. 

https://github.com/o1-labs/o1js/blob/a8b9b5e761599a8168ec858b3671ffbd1e771fe0/src/lib/mina/actions/reducer.ts#L70